src/pygeoprocessing/geoprocessing.py

```diff
@@ -2455,19 +2455,31 @@ def convolve_2d(
             to ``signal_path_band``'s nodata value or signal pixels where the
             kernel extends beyond the edge of the raster are not included when
             averaging the convolution filter. This has the effect of
-            "spreading" the result as though nodata and edges beyond the bounds
-            of the raster are 0s. If set to false this tends to "pull" the
-            signal away from nodata holes or raster edges. Set this value
+            "spreading" the result as though nodata and edges beyond the
+            bounds of the raster are 0s. If set to false this tends to "pull"
+            the signal away from nodata holes or raster edges. Set this value
             to ``True`` to avoid distortions signal values near edges for
             large integrating kernels.
+                It can be useful to set this value to ``True`` to fill
+            nodata holes through distance weighted averaging. In this case
+            ``mask_nodata`` must be set to ``False`` so the result does not
+            mask out these areas which are filled in. When using this
+            technique be careful of cases where the kernel does not extend
+            over any areas except nodata holes, in this case the resulting
+            values in these areas will be nonsensical numbers, perhaps
+            numerical infinity or NaNs.
         normalize_kernel (boolean): If true, the result is divided by the
             sum of the kernel.
         mask_nodata (boolean): If true, ``target_path`` raster's output is
             nodata where ``signal_path_band``'s pixels were nodata. Note that
             setting ``ignore_nodata_and_edges`` to ``True`` while setting
-            ``mask_nodata`` to False would be a nonsensical result and would
-            result in exposing the numerical noise where the nodata values were
-            ignored. An exception is thrown in this case.
+            ``mask_nodata`` to ``False`` can allow for a technique involving
+            distance weighted averaging to define areas that would otherwise
+            be nodata, however be careful in cases where the kernel does not
+            extend over any valid non-nodata area since the result can be
+            numerical infinity or NaNs. would be a nonsensical result and
+            would result in exposing the numerical noise where the nodata
+            values were ignored. An exception is thrown in this case.
```

Comment:
Is this last part left over?
>would be a nonsensical result and would result in exposing the numerical noise where the nodata values were ignored. An exception is thrown in this case.

Reply:
Oops, it sure was! Fixed in 6a727e32a4866d5a83ecdb6e1c78f25034bfb9c1